oneview_redfish_toolkit/blueprints/vlan_network_interface.py

```diff
@@ -0,0 +1,114 @@
+from oneview_redfish_toolkit.api.vlan_network_interface \
+    import VLanNetworkInterface
+from oneview_redfish_toolkit.api.vlan_network_interface_collection \
+    import VLanNetworkInterfaceCollection
+from oneview_redfish_toolkit.blueprints.util.response_builder \
+    import ResponseBuilder
+
+vlan_network_interface = Blueprint("vlan_network_interface", __name__)
+
+
+@vlan_network_interface.route(
+    ResourceBlock.BASE_URI +
+    "/<uuid>/EthernetInterfaces/<id>/VLANs", methods=["GET"])
+@vlan_network_interface.route(
+    ComputerSystem.BASE_URI +
+    "/<uuid>/EthernetInterfaces/<id>/VLANs", methods=["GET"])
+def get_vlan_network_interface_collection(uuid, id):
+    """Get Redfish VLanNetworkInterfaceCollection for the given UUID and ID.
+
+        Return VLanNetworkInterfaceCollection redfish JSON for a given
+        ServerProfileTemplate UUID and connection ID.
+        Logs exception of any error and return Internal Server
+        Error or Not Found.
+
+        Returns:
+            JSON: Redfish json with VLanNetworkInterfaceCollection.
+    """
+
+    try:
+        server_profile_template = \
+            g.oneview_client.server_profile_templates.get(uuid)
+
+        connSettings = server_profile_template["connectionSettings"]
+        connection = None
+
+        for conn in connSettings["connections"]:
+            if str(conn["id"]) == id:
+                connection = conn
+                break
+
+        if not connection:
+            raise OneViewRedfishError("Ethernet interface not found")
+
+        network_set = \
+            g.oneview_client.network_sets.get(connection["networkUri"])
+
+        vlan_collection = \
+            VLanNetworkInterfaceCollection(network_set, request.path)
+
+        return ResponseBuilder.success(vlan_collection)
+
+    except OneViewRedfishError as e:
+        # In case of error log exception and abort
+        logging.exception('Unexpected error: {}'.format(e))
+        abort(status.HTTP_404_NOT_FOUND, e.msg)
+
+
+@vlan_network_interface.route(
+    ResourceBlock.BASE_URI +
+    "/<uuid>/EthernetInterfaces/<id>/VLANs/<vlan_id>", methods=["GET"])
+@vlan_network_interface.route(
+    ComputerSystem.BASE_URI +
+    "/<uuid>/EthernetInterfaces/<id>/VLANs/<vlan_id>", methods=["GET"])
+def get_vlan_network_interface(uuid, id, vlan_id):
+    """Get the Redfish VLanNetworkInterface for the given UUID, ID and VLanID.
+
+        Return VLanNetowrkInterface redfish JSON for a given
+        ServerProfileTemplate UUID, EthernetInterface ID and a VLan ID.
+        Logs exception of any error and return Internal Server
+        Error or Not Found.
+
+        Returns:
+            JSON: Redfish json with ResourceBlock.
+    """
+
+    try:
+        ethernet_network = \
+            g.oneview_client.ethernet_networks.get(vlan_id)
+
+        vlan = VLanNetworkInterface(ethernet_network, request.path)
+
+        return ResponseBuilder.success(vlan)
+
+    except OneViewRedfishError as e:
```

Comment:
Can OneViewRedfishError type exception occur here? I'm asking because it doesn't seem be raised in this method, because if any exception happens, flask will catch through response builder.

Reply:
Done.